### PR TITLE
Introduce presto adapter

### PR DESCRIPTION
**Pull request by snowhork**
This PR is for support Presto.

To support Presto, this PR includes some features as below:

* allow raw_datasets.count nil (presto's datastore is hard to count up)
* ignored_tables also ignore schema (some schema should be ignored in presto's datastore)
* add PrestoAdapter class

please review @hogelog

### CODE REVIEW

**Comment by snowhork** [comment]
Screen shots:

image

image

File: app/models/data_source.rb [hunk 1]
```diff
@@ -44,4 +40,14 @@ def data_source_adapter
   def standard_adapter?
     adapter.in? DataSourceAdapters.standard_adapter_names
   end
+
+  private
+
+  def ignore(schema_name, table_name)
+    ignored_table_patterns.match("#{schema_name}.#{table_name}")
+  end
```
Comment (hogelog) [inline app/models/data_source.rb]: I would like to name `ignore?` to this method.

File: app/models/data_source.rb [hunk 2]
```diff
@@ -12,11 +12,11 @@ class ConnectionBad < IOError
 
   def source_table_names
     table_names = access_logging { data_source_adapter.fetch_table_names }
-    table_names.reject { |_, table_name| ignored_table_patterns.match(table_name) }
+    table_names.reject { |schema_name, table_name| ignore(schema_name, table_name) }
   end
 
   def data_source_table(schema_name, table_name, table_names)
-    return if ignored_table_patterns.match(table_name)
+    return if ignore(schema_name, table_name)
```
Comment (hogelog) [inline app/models/data_source.rb]: > __[RuboCop]__ Layout/EmptyLineAfterGuardClause: Add empty line after guard clause. ([link](https://sider.review/gh/repos/55787616/pulls/112#issue-3544833))

I don't like this rule.

Comment (snowhork) [reply app/models/data_source.rb]: b8a39e0

**Comment by snowhork** [comment]
Changelog is added, so please rereview @hogelog .

**Pull request merged**